src/Avalonia.Base/Data/Core/Plugins/MethodAccessorPlugin.cs

```diff
@@ -55,13 +55,20 @@ public class MethodAccessorPlugin : IPropertyAccessorPlugin
 
             var methods = type.GetMethods(bindingFlags);
 
-            foreach (MethodInfo methodInfo in methods)
+            foreach (var methodInfo in methods)
             {
                 if (methodInfo.Name == methodName)
                 {
-                    found = methodInfo;
-
-                    break;
+                    var parameters = methodInfo.GetParameters();
+                    if (parameters.Length == 1 && parameters[0].ParameterType == typeof(object))
+                    {
+                        found = methodInfo;
+                        break;
+                    }
+                    else if (parameters.Length == 0)
+                    {
+                        found = methodInfo;
+                    }
```

Comment:
should there be a break; here

Reply:
I don't think so, that would keep the bug, as the 1 parameter overload is preferred.